module/apmotel/wrapper.go

```diff
@@ -0,0 +1,68 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package apmotel // import "go.elastic.co/apm/module/apmotel/v2"
+
+import (
+	"context"
+
+	"go.elastic.co/apm/v2"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+func init() {
+	// We override the apm context functions so that spans started with the
+	// native API are wrapped and made available as OTel spans.
+	apm.OverrideContextWithSpan = contextWithSpan
+	apm.OverrideSpanFromContext = spanFromContext
+	apm.OverrideContextWithTransaction = contextWithTransaction
+	apm.OverrideTransactionFromContext = transactionFromContext
+}
+
+func contextWithSpan(ctx context.Context, s *apm.Span) context.Context {
+	// Should we make a note that they need to update the default tracer to
+	// use this? Or is there some better way to do this.
+	otelSpan := &span{inner: s, tracer: apm.DefaultTracer(), tx: transactionFromContext(ctx)}
+	return trace.ContextWithSpan(ctx, otelSpan)
+}
```

Comment:
I went through the apmot code again and now I recall why there's multiple Transaction and Span methods.

- `contextWithSpan` adds an `*apm.Span` (not an OpenTracing span) to context. Similar for `contextWithTransaction`
- `spanFromContext` returns an `*apm.Span` (not an OpenTracing span) from context. Similar for `transactionFromContext`

When you call `apm.ContextWithSpan` with an `*apm.Span` (e.g. when you call `apm.StartSpan`), it will call `contextWithSpan`, which wraps the `*apm.Span` in a struct that makes it comply to the `opentracing.Span` interface. This is then added to context using `opentracing.ContextWithSpan`.

Let's say you then call `opentracing.StartSpanFromContext`: this extracts the wrapper span from context to fetch the parent SpanContext, creates a new `otSpan`, and then adds that to context. The `otSpan` type also has a `Span` method so `apm.SpanFromContext` can extract the underlying `*apm.Span`.

This way you can mix both `apm.Spans` and `opentracing.Spans` together in the same context. Long story short: I think we need to be doing something similar here.